docs(mdbook): rewrite chiplets design description

grjte opened the pull request:
- [x] rewrite description of chiplets module
- [x] move bitwise selector constraints to bitwise
- [x] update chiplets execution trace diagram
- [x] shorten & update description of "last row problem" to include the recent adjustments made to memory
- [x] update the description of the transition label `m` in the hasher permutation check descriptions to match the latest changes to $b_{chip}$

bobbinth commented on docs/src/design/chiplets/hasher.md:
```diff
@@ -363,7 +363,7 @@ $$
 In the above:
 
-- $m$ is a _transition label_ which uniquely identifies each transition function.
+- $m$ is a _transition label_, composed of the [operation label](main.md#operation-labels) and the periodic columns, which uniquely identify each transition function.
```
It might be good to add one or two sentences here explaining why we mix in periodic columns into the label.

grjte replied:
I've added a bit more explanation. Please let me know what you think

bobbinth reviewed (changes requested):
Looks great! Thank you! I've left a few small comments inline.

grjte commented on docs/src/design/chiplets/main.md:
```diff
@@ -1,196 +1,122 @@
-  $$r' - r - 1 = 0$$
-- [Memory](./memory.md) - all transition constraints, in particular this degree 6 constraint:
-  $$(1−n_0)⋅(n_1^2−n_1)=0$$
+**In the memory chiplet:** all transition constraints cause a conflict. To adjust for this, the selector flag for the memory chiplet is designed to exclude its last row. Thus, memory constraints will not be applied when transitioning from the last row of the memory chiplet to the subsequent row. This is achieved without any additional increase in the degree of constraints by using $s'_2$ as a sel ...
```
I think you must have missed it... It's used in the part that shows how the flags are applied to each chiplets and it's mentioned in one of those bullet points with a link that comes back to this section. Let me know if you still want a change though

bobbinth replied:
ah yes! must not have fully woken up yet :)

grjte replied:
The constraints section is updated a bit now, which may make this clearer.

grjte commented on docs/src/design/chiplets/main.md:
```diff
@@ -1,196 +1,122 @@
+s_0^2 - s_0 = 0 \\
+s_0 \cdot (s_1^2 - s_1) = 0 \\
+s_0 \cdot s_1 \cdot (s_2^2 - s_2) = 0
+$$
```
Updated, thanks!

grjte commented on docs/src/design/chiplets/bitwise.md:
```diff
@@ -6,15 +6,15 @@ Assume that $a$ and $b$ are field elements in a 64-bit prime field. Assume also
-$$
+> $$
 xor(a, b) = a + b - 2 \cdot a \cdot b
 $$
```
Fixed!

grjte commented:
Thanks @bobbinth! I did a full pass of all the chiplets and made some more updates to constraint descriptions on the Bitwise page. I also added one set of missing constraints ro the main Chiplets module description and updated the whole constraints section for clarity. Please let me know if there's anything else!

bobbinth reviewed (approved):
All looks good! Thank you!

We'll need to revisit individual chiplet sections (i.e., hasher, bitwise, memory) to make sure everything there is consistent with the latest approach to describing constraints (e.g., no references to $\beta$) - but this is for another PR.
